src/DataWrangling/ECCO/ECCO_darwin.jl

```diff
@@ -0,0 +1,179 @@
+using MeshArrays
+using JLD2
+using Glob
+
+struct ECCO2DarwinMonthly <:SomeECCODataset end
+struct ECCO4DarwinMonthly <:SomeECCODataset end
+
+# URLs for the ECCO datasets specific to each version
+const ECCO4Darwin_url = "https://ecco.jpl.nasa.gov/drive/files/ECCO2/LLC90/ECCO-Darwin/"
+const ECCO2Darwin_url = "https://ecco.jpl.nasa.gov/drive/files/ECCO2/LLC270/ECCO-Darwin_extension/"
+
+Base.size(data::Metadata{<:ECCO4DarwinMonthly}) = (720,  360, 50, length(data.dates))
+Base.size(::Metadatum{<:ECCO4DarwinMonthly})    = (720,  360, 50, 1)
+Base.size(data::Metadata{<:ECCO2DarwinMonthly}) = (1440, 720, 50, length(data.dates))
+Base.size(::Metadatum{<:ECCO2DarwinMonthly})    = (1440, 720, 50, 1)
+
+# The whole range of dates in the different dataset datasets
+all_dates(::ECCO4DarwinMonthly, name) = DateTime(1992, 1, 1) : Month(1) : DateTime(2023, 3, 1)
+all_dates(::ECCO2DarwinMonthly, name) = DateTime(1992, 1, 1) : Month(1) : DateTime(2025, 5, 1)
+
+ECCO_Darwin_timestep(::Metadatum{<:ECCO4DarwinMonthly}) = 3600
+ECCO_Darwin_timeref(::Metadatum{<:ECCO4DarwinMonthly}) = DateTimeProlepticGregorian(1992, 1, 1, 12, 0, 0)
+
+ECCO_Darwin_timestep(::Metadatum{<:ECCO2DarwinMonthly}) = 1200
+ECCO_Darwin_timeref(::Metadatum{<:ECCO2DarwinMonthly}) = DateTimeProlepticGregorian(1992, 1, 1, 0, 0, 0)
+
+# File name generation specific to each Dataset dataset
+function metadata_filename(metadata::Metadatum{<:Union{ECCO2DarwinMonthly, ECCO4DarwinMonthly}})
+    shortname = dataset_variable_name(metadata)
+    
+    reference_date = ECCO_Darwin_timeref(metadata)
+    timestep_size  = ECCO_Darwin_timestep(metadata)
+
+    # Explicitly convert to Int to avoid return of a float
+    iternum = Int(Dates.value((metadata.dates - reference_date) / (timestep_size * 1e3)))
+    iterstr = string(iternum, pad=10)
+
+    return shortname * "." * iterstr * ".data"
+end
+
+# Convenience functions
+default_mask_value(::ECCO4DarwinMonthly) = 0
+default_mask_value(::ECCO2DarwinMonthly) = 0
+
+dataset_variable_name(data::Metadata{<:Union{ECCO2DarwinMonthly,ECCO4DarwinMonthly}}) = ECCO_darwin_dataset_variable_names[data.name]
+
+location(::Metadata{<:Union{ECCO2DarwinMonthly, ECCO4DarwinMonthly}}) = (Center, Center, Center)
+
+variable_is_three_dimensional(::Metadata{<:Union{ECCO2DarwinMonthly, ECCO4DarwinMonthly}}) = true
+
+ECCO_darwin_dataset_variable_names = Dict(
+    :temperature => "THETA",
+    :salinity    => "SALTanom",
+    :DIC => "DIC",
+    :ALK => "ALK",
+    :PO₄ => "PO4",
+    :NO₃ => "NO3",
+    :DOP => "DOP",
+    :POP => "POP",
+    :Fe  => "FeT",
+    :Siᵀ => "SiO2",
+)
+
+ECCO_darwin_scale_factor = Dict(
+    :temperature => 1,
+    :salinity    => 1,
+    :DIC => 1e-3,
+    :ALK => 1e-3,
+    :PO₄ => 1e-3,
+    :NO₃ => 1e-3,
+    :DOP => 1e-3,
+    :POP => 1e-3,
+    :Fe  => 1e-3,
+    :Siᵀ => 1e-3,
+)
```

Comment:
can you include a comment about what this is for?

Reply:
It's a unit conversion.

Reply:
Commented

Reply:
I got rid of these conversion dicts in favor of mimicking the `convert_temperature` routine in `metadata_field` - `convert_concentration` now does this more cleanly, and can be applied to other datasets.